### PR TITLE
usb: hub: doc: Fix Doxygen documentation

### PR DESCRIPTION
**Pull request by kartben**
Create proper sections and add missing Doxygen comments.

### CODE REVIEW

File: include/zephyr/usb/class/usb_hub.h [hunk 1]
```diff
@@ -12,33 +12,41 @@
...
+#define USB_HCFS_PORT_RESET		0x04    /**< Reset */
+#define USB_HCFS_PORT_POWER		0x08    /**< Port Power */
+#define USB_HCFS_PORT_LOW_SPEED		0x09    /**< Low-speed device attached */
+#define USB_HCFS_C_PORT_CONNECTION	0x10    /**< Connect Status Change */
+#define USB_HCFS_C_PORT_ENABLE		0x11    /**< Port Enable/Disable Change */
+#define USB_HCFS_C_PORT_SUSPEND		0x12    /**< Suspended Change */
+#define USB_HCFS_C_PORT_OVER_CURRENT	0x13    /**< Over-current Indicator Change */
+#define USB_HCFS_C_PORT_RESET		0x14    /**< Reset Change */
+#define USB_HCFS_PORT_TEST		0x15    /**< Port Test Mode */
+#define USB_HCFS_PORT_INDICATOR		0x16    /**< Port Indicator Control */
```
Comment (jfischer-no) [inline include/zephyr/usb/class/usb_hub.h:34]: These comments are completely unnecessary, the names are self-explanatory.  This should be rather internally hidden.

**Review by jfischer-no — CHANGES_REQUESTED** [review]

File: include/zephyr/usb/class/usb_hub.h (hunk 1 above)
Comment (tmon-nordic) [reply include/zephyr/usb/class/usb_hub.h:34]: There is no official "short description" for these as the specification just uses the identifiers (without the `USB_HCFS_` prefix) directly.

Comment (kartben) [reply include/zephyr/usb/class/usb_hub.h:34]: > There is no official "short description" for these as the specification just uses the identifiers (without the `USB_HCFS_` prefix) directly.

I actually took these verbatim from the specification, as they are "spelled out" in 11.24.2.7.1 and 11.24.2.7.2.

Comment (tmon-nordic) [reply include/zephyr/usb/class/usb_hub.h:34]: OK, I see. You took the bold text as description.

**Review by gmarull — APPROVED** [review]
Some defines are self-explanatory, yes, but we need a consistent Doxygen output. To me, this is a project-wide decision, not up to maintainers to discuss.

File: include/zephyr/usb/class/usb_hub.h (hunk 1 above)
Comment (carlescufi) [reply include/zephyr/usb/class/usb_hub.h:34]: @jfischer-no see comments from @kartben

**Review by tmon-nordic — APPROVED** [review]
I think this shouldn't really be public header, but given that it currently is, I am not objecting this.

**Review by jfischer-no — CHANGES_REQUESTED** [review]
It should not be a public header for user/application. NACK to documentation style.

**Pull request closed without merging**